JavaScript: arguments with object contains newline in arrow function

sosukesuzuki opened the pull request:
<!-- Please provide a brief summary of your changes: -->
Fix #6374 
<!-- Please ensure you’ve done all of these things (if applicable). -->
<!-- You can replace the `[ ]` with `[x]` to mark each task as done. -->

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

evilebottnawi commented:
/cc @sosukesuzuki need rebase

/cc @prettier/core

thorn0 commented on CHANGELOG.unreleased.md:
````diff
@@ -392,6 +392,39 @@ const foo = [abc, def, ghi, jkl, mno, pqr, stu, vwx, yz] as (
 )[];
 ```
 
+#### JavaScript: Fix breaks indentation arguments with nested object contains newline in arrow function ([#6382] by [@sosukesuzuki])
````
This PR does need a separate changelog entry. It can share one with #6301.

thorn0 commented on src/language-js/printer-estree.js:
```diff
@@ -4075,16 +4075,21 @@ function printArgumentsList(path, options, print) {
+  function hasEmptyLineInObject(node) {
+    return (
+      node &&
+      node.type &&
```
This line can be omitted because you have `node.type === "ObjectPattern"` on the next line.

thorn0 commented:
**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACAklADgVxgB0oAzCCAChIw0oG1gbaMBDAGhOdoCNmBfALoBKDAF4AfBmD8SwgNwh2ICLhgBLaAGdkoVgCd9EAO4AFAwh0pWANwjqAJkpA99rMAGs4MAMq536lAA5sgw+vhwygAWMAC2ADYA6lHq8Fr+YHA+lqnqNqkAnsjgWjrKgVpw+jCmbkGxrMikrPGVygBWWgAeAEJunt4+rLFwADKBcE0tbSCdXT6BQfFwAIr4EPBTrZEg-vqV+sU8rDwF8dDOuPqBMImOMFHIABwADMpXEJWJbrjFV3AHGyTZQAR3W8FqqisIFYWgAtFA4HAHMjnPo4GD1Ojaqx6o0kM1tspKrF1KFwjstItlmsNpMCdMdjATncHA9kAAmZRhVjqeKLADCEFiDWKUGgwJA+EqABUTlZCTMbBEcCjYD4wNc1ABBKAOHwwM70xVwfj8IA)
```sh
--parser babylon
```

**Input:**
```jsx
// Input
foo(
  ([{
    a,

    b
  }]) => {}
);
```

**Output:**
```jsx
// Input
foo(([{ a,
    b }]) => {});

```

sosukesuzuki commented:
@thorn0 ( /cc @evilebottnawi ) Should I fix it in this PR? or in another PR?

thorn0 commented:
It's the same issue. Would be logical if you fixed it here.

evilebottnawi commented:
@sosukesuzuki let's do it here :+1:

thorn0 commented:
Did you guys know this is valid syntax? 😮

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACAklADgVxgB0oAzCCAChIw0oG0A6Z4G2jAQwBoS3aAjNgF8AugEoMAXgB8GYEJJiA3CC4gIuGAEtoAZ2SgOAJyMQA7gAVjCfSg4A3CFoAmqkPyMcwAazgwAyrheWlAA5sgwRvhwagAWMAC2ADYA6rFa8LpBYHD+Nhla9hkAnsjguvpqIbpwRjAWnqEJHMikHEk1agBWugAeAEKePn7+HAlwADIhcK3tnSA9vf4hoUlwAIr4EPCzHTEgQUY1RmX8HPzFSdBuuEYhMCkuMLHIABwADGq3EDUpnrhlW5wY72GZqACOW3gDQ0thAHF0AFooHA4M40W4jHBIVosQ0OE0Wkg2ns1DUEloIlF9roVmtNtsZsS5vsYOdHs5nsgAExqSIcLRJFYAYQgCWaZSg0DBIHwNQAKudbCT5vZojh0bB-GA7poAIJQZz+GCXJkquBCIRAA)
```sh
--parser babylon
```

**Input:**
```jsx
// Input
foo(
  ([...{
    a,

    b
  }]) => {}
);
```

**Output:**
```jsx
// Input
foo(([...{ a,
    b }]) => {});

```

sosukesuzuki commented:
Thanks.
Also, I noticed this is valid.

[playground](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAQzwF49hcSTSAaXWuvAIxYIF8WBKSgHzUeUXgG5cIBiAgAHGAEtoAZ2ShSAJw0QA7gAVNCVSlIA3CAoAmUkGw2kwAazgwAyrIcKoAc2QwNAK5w0gAWMAC2ADYA6iEK8MoeYHCuRvEKpvEAnsjgyqrSXspwGjB69t7hpMiopJHF0gBWygAeAEL2Ti6upOFwADJecDV1DSDNLa5e3pFwAIoBEPAj9cEgHhrFGrlspGxwkTayGl4w0VYwIcgAHAAM0scQxdH2srnHcFumw9IAjovwcpyYwgUjKAC0UDgcEsMJsGjg-wUCPKpEq1SQtVW0mK4QUfkCa2U01mCyWw0xozWMD250sl2QACZpP5SApItMAMIQcJVXJQaA-EABYoAFT2xixY1MQQAklBYbBXGATvIAIIK1wwLKzFbFLhcIA)

```js
// Input
foo(
  (
    a = {
      a,

      b
    }
  ) => {}
);

// Output
foo((a = { a,
  b }) => {});
```

evilebottnawi commented on src/language-js/printer-estree.js:
```diff
@@ -4075,14 +4075,30 @@ function printArgumentsList(path, options, print) {
+          (param.type === "AssignmentPattern" &&
+            param.right &&
+            hasEmptyLineInObject(param.right))
+      )
```
maybe we can move this to function, for example hasNewlineInParams?

sosukesuzuki replied:
Where should I move the function? Also, [this parts](https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L4092-L4112) is related?

evilebottnawi replied:
just question :smile:

sosukesuzuki replied:
Oh, sorry.
I might be able to move it. But, I think I should not do it in this pr.
Probably, the range of influence will be larger.

thorn0 commented:
Still doesn't work for nested cases like this:

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACAklADgVxgB0oAzCCAChIw0uAA8kBtYG2jAQwBoT3aARuwC+AXWEBKDAF4AfBmDCSEgNwhuICLhgBLaAGdkoTgCcTEAO4AFUwkMpOANwg6AJupACTnMAGs4MADKuD46UADmyDAm+HAaABYwALYANgDq8Trw+iFgcIF2WTqOWQCeyOD6hhph+nAmMFbe4UmcyKScKXUaAFb6DABC3n4BgZxJcAAyYXDtnd0gfQyBYeEpcACK+BDwc11xICEmdSYVApwCpSnQHrgmYTBpbjDxyAAcAAwadxB1ad64Cp3OAnRyzDQAR228CaWnsIE4+gAtFA4HBXOiPCY4FCdNimpwWm0kB19ho6kkdFEYgd9Kt1lsdrMSfMDjALk9XC9kAAmDTRTg6FKrADCECSrQqUGg4JA+DqABULvZSQtHLEcBjYIEwPdtABBKCuQIwK7M1VwYTCIA)
```sh
--parser babylon
```

**Input:**
```jsx
// Input
foo(
  ({x:[{
    a,

    b
  }]}) => {}
);
```

**Output:**
```jsx
// Input
foo(({ x: [{ a,
      b }] }) => {});

```

sosukesuzuki commented:
Oh..
Also, doesn't work this case too...

[Playground Link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACAklADgVxgB0oAzCCAChIw2qltoEMMBeDAbRsduG5+YAaEvwEAjUbQC+ogLrcAlGwB8GYDKgKA3CRCCQEXDACW0AM7JQTAE7WIAdwAKNhBZRMAbhGMATPSDFrJjAAazgYAGVcYOMoAHNkGGt8OH0ACxgAWwAbAHU043gzaLA4CNdC4w9CgE9kcDMLfVizOGsYRyC4zKZkUiZs1v0AKzMADwAhINDwiKZMuAAZWLg+gaGQUbGI2LjsuABFfAh4NcHUkGjrVut6sSYxGuzof1xrWJhc3xg05AAOAAM+jeEFauSCuHqbzgNw8q30AEdjvBOoY3CAmGYALRQOBwHz4-zWOBI4zEzpMbq9JD9c76VqZYyJZIXMy7fZHE6rGnrC4wB5fHw-ZAAJn0SSYxmyuwAwhBMj16lBoPCQPhWgAVB5uWkbDwpHAE2ARMDvIwAQSgPgiMCe3N1cCkUiAA)

```js
// Input
foo(
  (
    a = [
      {
        a,

        b
      }
    ]
  ) => a
);

// Output
foo((a = [{ a,
    b }]) => {});
```

thorn0 commented:
I'm truly sorry... 😇

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEB6VACAklADgVxgB0oAzCCAChIw0oG17gbaMBDAGhJdoCMWAvgF0hASgwBeAHwZgAkqIDcIDiAi4YAS2gBnZKDYAnQxADuABSMI9KNgDcImgCYqQvQ2zABrODADKuJ6aUADmyDCG+HCqABYwALYANgDqMZrwOoFgcH7W6Zp26QCeyOA6eqrBOnCGMOYeIfFsyKRsidWqAFY6AB4AQh7evn5s8XAAMsFwLW0dIN09fsEhiXAAivgQ8DPt0SCBhtWGpbxsvEWJ0K64hsEwyc4wMcgAHAAMqjcQ1ckeuKU3OBHOzTVQAR028Hq6hsIDYOgAtFA4HAnKjXIY4BDNJj6mxGs0kK1dqpqvFNOFInsdMtVhsttMibM9jAzg8nE9kAAmVQRNiaRLLADCEHiTVKUGgoJA+GqABUzjZiXM7FEcGjYH4wLcNABBKBOPwwC6M5VwAQCIA)
```sh
--parser babylon
```

**Input:**
```jsx
// Input
foo(
  ([[{
    a,

    b
  }]]) => {}
);
```

**Output:**
```jsx
// Input
foo(([[{ a,
      b }]]) => {});

```

thorn0 commented:
Sorry, but I've found a couple more cases.

1. Rest parameter combined with object pattern:

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPTAOhOAEMAaXAm2622gIwF8BKPAXgD49hndWAbhAUQEAA4wAltADOyUGQBOSiAHcACsoTyUZAG4QpAExEhGSsmADWcGAGVxVqVADmyGEoCucUQAsYAFsAGwB1Pyl4WScwOHsdSKl9SIBPZHBZeVEXWTglGA1LV0CyZFQyYNzRACtZAA8AIUsbO3syQLgAGRc4MoqqkFq6+xdXYLgARS8IeD7K3xAnJVyldMYyRhTg6DNxJRcYUJMYP2QADgAGUT2IXNDLcXS9uBX9XtEAR2n4QoldEDIsgAtFA4HBjOCzEo4F8pNDCmRiqUkOV5qJcoEpB5vAtZKNxlMZr0Uf0FjANkdjCdkAAmUSeMhSYKjADCEECJXSUGg7xAXlyABUNrpUQN9D4AJJQCGwexgfaSACC0vsMC2xNFcGYzCAA)
```sh
--parser babylon
```

**Input:**
```jsx
foo(
  (...{a,
         
         b}) => {}
);
```

**Output:**
```jsx
foo((...{ a,
  b }) => {});

```

2. Rest parameter combined with array pattern. This one seems really strange. I can't imagine how it can be useful, but still you might want two fix it.

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPTAOhIG1gBDAGlwLvtvvoCMBfAXQEo8BeAPjzBWuTgG4QVEBAAOMAJbQAzslAUATmogB3AArqEylBQBuEOQBMJIZmopgA1nBgBlaXblQA5shhqArnCSABYwALYANgDqQXLwim5gcM4GsXLGsQCeyOCKypIeinBqMDq2nqEUyKgU4YWSAFaKAB4AQrYOTs4UoXAAMh5wVTV1II1Nzh6e4XAAin4Q8EO1gSBuaoVq2cwUzBnh0FbSah4wkRYwQcgAHAAMkkcQhZG20tlHcBvGg5IAjvPwpRkhhAFEUAFooHA4OZoVY1HA-nJ4aUKOVKkhqstJIVQnIfP4VopJtM5gtBhjhisYDszuYLsgAEySXwUOThSYAYQgoQq2Sg0G+ID8hQAKjtDJiRsYAgBJKAw2DOMDHWQAQXlzhge3JkrgrFYQA)
```sh
--parser babylon
```

**Input:**
```jsx
foo(
  (...[{a,
         
         b}]) => {}
);
```

**Output:**
```jsx
foo((...[{ a,
    b }]) => {});

```

3. Even more strange and (seemingly?) useless one.

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPTAbQDoyjgBDAGlwPoboYYCMBfAXQ4Eo8BeAHx5gbXNwDcIaiAgAHGAEtoAZ2ShKAJw0QA7gAVNCVSkoA3CAoAmUkCw2UwAazgwAyrIcKoAc2QwNAK5w0gAWMAC2ADYA6iEK8MoeYHCuRvEKpvEAnsjgyqrSXspwGjB69t7hlMiolJHF0gBWygAeAEL2Ti6ulOFwADJecDV1DSDNLa5e3pFwAIoBEPAj9cEgHhrFGrkslCxZkdA2shpeMNFWMCHIABwADNInEMXR9rK5J3BbpsPSAI6LeDlOTGECUZQAWigcDgllhNg0cABCkR5UolWqSFqq2kxXCCj8gTWymmswWS2GWNGaxgewuliuyAATNJ-JQFJFpgBhCDhKq5KDQX4gALFAAqe2M2LGpiCAEkoHDYK4wKd5ABBRWuGAHSnSuBsNhAA)
```sh
--parser babylon
```

**Input:**
```jsx
foo(
  ([...[{a,
         
         b}]]) => {}
);
```

**Output:**
```jsx
foo(([...[{ a,
      b }]]) => {});

```

sosukesuzuki commented:
I found a case, too

[Playground](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAQzwF48BtXEk4O+k0gGlyebwCNOSBfPgF0mASkoA+PMEFRRAblwhWICAAcYAS2gBnZKFIAnQxADuABSMI9KUgDcImgCbKQ3Q6TABrODADKap6aUADmyDCGAK5wKgAWMAC2ADYA6rGa8DqBYHB+1hmadhkAnsjgOnoqwTpwhjDmHiEJpMiopEk1KgBWOgAeAEIe3r5+pAlwADLBcK3tnSA9vX7BIUlwAIqREPCzHTEggYY1hmXcpNxwSa5qhsEwKc4wscgAHAAMKjcQNSkeamU3ODHOwzFQARy28Aa6hsIFIOgAtFA4HAnKjXIY4BDNJiGqQmi0kG09ioaglNOEovsdCs1pttjMiXN9jBzg8nE9kAAmFQRUiaJIrADCEASzTKUGgoJAkRqABVzjZifM7NEAJJQNGwPxgW4aACCmr8MGKa12NX4-CAA)

```js
// Input
foo(
  (
    a = [
      {
        a,

        b
      }
    ]
  ) => {}
);

// Output
foo((a = [{ a,
    b }]) => {});
```

thorn0 commented:
Keep in mind that there can be any expression after `=`, e.g. a function call:

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAQzwF49cSTVNgbaTSAaXJ5vAI05IF8AlE0GUAfHmD9cggNy4QrEBAAOMAJbQAzslCkATvogB3AAoGEOlKQBuEdQBNFIbvtJgA1nBgBlFe-UoAHNkGH0AVzglAAsYAFsAGwB1aPV4LX8wOB9LNPUbNIBPZHAtHSVArTh9GFM3ILjSZFRSBKqlACstAA8AITdPbx9SOLgAGUC4Ztb2kC7un0CghLgARXCIeGm2qJB-fSr9Eu5SbjgE5xV9QJgkxxho5AAOAAYlK4gqpLcVEqu4Q42KZKACOG3gdVUVhApC0AFooHA4A5kc59HAwep0XVSA0mkgWjslFU4upQhFdlolit1pspgSZrsYKc7g4HsgAExKMKkdQJJYAYQgcUaJSg0GBIHCVQAKqcrITZjZIgBJKAo2A+MDXNQAQXVPhghRW2yq-H4QA)
```sh
--parser babel
```

**Input:**
```jsx
foo(
  (
    a = 
      f({
        a,

        b
      })
  ) => {}
);

```

**Output:**
```jsx
foo((a = f({ a,
  b })) => {});

```

sosukesuzuki commented:
I found 2 cases

###  pass `ArrowFunctionExpression` to `AssignmentPattern`

[Playground Link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAQzwF5DhcSTSAaXWuvAIxYIF8BKSgPjzAuLPhUHDcPANy4QDEBAAOMAJbQAzslCkATrogB3AAp6EWlKQBuEVQBN5INrtJgA1nBgBlJa9VQAc2QYXQBXOAUACxgAWwAbAHVI1XgNXzA4L3MU1SsUgE9kcA0tBX8NOF0YYxcAmNJkVFI4ioUAKw0ADwAhF3dPL1IYuAAZfzhG5taQDs6vfwC4uABFUIh4SZaIkF9dCt0itlI2ODjHJV1-GAT7GEjkAA4ABgULiAqElyUii7h9qwmCgAjmt4DVlBYQKQNABaKBwOB2RGOXRwEGqVE1Uh1BpIJpbBQVGKqYJhbYaBZLVbrCZ4qbbGDHG52O7IABMChCpFUcQWAGEIDF6kUoNBASBQhUACrHCz46ZWcIASSgSNgXjAlxUAEFVV4YPklpsKlwuEA)

```js
// Input
foo(
  (
    a = ({
      a,

      b
    }) => {}
  ) => {}
);

// Output
foo((a = ({ a,
  b }) => {}) => {});
```

### Pass `CallExpression` (IIFE) to `AssignmentPattern`.

[Playground Link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAQzwF5DNhcSTSAaXO+vAI1YIF8BKSgHx5gfTL1b8KQkbl4BuXCEYgIABxgBLaAGdkoUgCcDEAO4AFQwl0pSANwgaAJkpDsDpMAGs4MAMqqPDSgAc2QYAwBXOGUACxgAWwAbAHUYjXhtALA4Xyt0jVt0gE9kcG1dZSDtOAMYM3dg+NJkVFJE6uUAK20ADwAhdy8fX1J4uAAZILgWto6Qbp7fIODEuABFCIh4GfbokACDaoNS9lJ2OESXVQMgmGSnGBjkAA4ABmVriGrk91VS67gR1s02UAEdNvB6mprCBSNoALRQOBwRwolwGODgjQY+qkRrNJCtXbKarxDRhSJ7bTLVYbLbTQmzPYwM73RyPZAAJmU4VIGkSywAwhB4k1SlBoCCQBFqgAVM7WIlzWxRACSUFRsF8YBu6gAghrfDAiqsdtVuNwgA)

```js
// Input
foo(
  (
    a = (({
      a,

      b
    }) => {})()
  ) => {}
);

// Output
foo((a = (({ a,
  b }) => {})()) => {});
```

thorn0 commented:
There can follow _any_ expression after `=`. Function calls were just an example.

**Prettier pr-6382**
[Playground link](https://deploy-preview-6382--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAQzwF49cSSBGAalU2BtpNIBpc328AjXiQC+ASjajKAPjzBhuUQG5cITiAgAHGAEtoAZ2ShSAJ2MQA7gAUTCAylIA3CNoAmqkP2OkwAazgwAZQ1vbSgAc2QYYwBXODUACxgAWwAbAHV47Xg9YLA4ANss7QcsgE9kcD0DNVC9OGMYSy8wpNJkVFIUurUAKz0ADwAhL19-ANIkuAAZULh2zu6QPv6A0LCUuABFaIh4ea64kGDjOuMK-lJ+OBT3DWNQmDTXGHjkAA4ABjU7iDq0rw0FTucFODjmagAjjt4E1NHYQKQ9ABaKBwOAudHuYxwKHabFNUgtNpIDoHNR1JLaSIxQ56NYbba7OYkhaHGCXJ4uF7IABMaiipG0KTWAGEIElWhUoNBwSBonUACqXOykxYOWIASSgGNgATA9y0AEFtQEYKUNvs6sJhEA)
```sh
--parser babel
```

**Input:**
```jsx
foo(
  (
    a = 
      1+f({
        a,

        b
      })
  ) => {}
);

```

**Output:**
```jsx
foo((a = 1 + f({ a,
    b })) => {});

```

sosukesuzuki commented:
@evilebottnawi @thorn0 
I deleted the previous implementation and added a new implementation that can handle common patterns. Please review again:pray:

sosukesuzuki commented on src/language-js/printer-estree.js:
```diff
@@ -4070,26 +4070,29 @@ function printArgumentsList(path, options, print) {
+      !arg.body ||
+      arg.body.type !== "BlockStatement" ||
+      !arg.params ||
+      arg.params.length < 1
```
I don't think we need to consider about it.
But, I found a new bug:
[Playground Link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCEAUAdKACPHfAvAelL3QlwIEo8BeAPj2AF9daBuXEAGhAQADjACW0AM7JQAQwBOciAHcACvIRSUMgG4RRAE34gARnJlgA1nBgBlIedFQA5shhyArnAEALGAFsAGwB1b1F4CXswOBsNMNFtMIBPZHAJKQFHCTg5GBUzJz8ZZFQZAKyBACsJAA8AITNLaxsZPzgAGUc4YtLykCrqm0cnALgARXcIeG6yrxB7OSy5FOMZYzgAoyE5RxgggxhvZAAOAAYBLYgsoLMhFK24Re0ugQBHCfg84U0QGQkAWigcDg+mBRjkcDeonBeRkBSKSBKMwEWT8olcHlmEiGI3Gky6CJ6sxgqz2+gOyAATAI3DJRAEhgBhCB+QopKDQZ4gdxZAAqq00iKybDYQA)
```js
// Input
foo(
  (
    // foo
  ) => {}
);

// Output
foo(() =>
  // foo
  {}
);
```

evilebottnawi replied:
Good catch, need create issue after merge

 /cc @prettier/core

thorn0 commented on src/language-js/printer-estree.js:
```diff
@@ -4070,26 +4070,29 @@ function printArgumentsList(path, options, print) {
+    let shouldBreak = false;
+    argPath.each(paramPath => {
+      const printed = concat([print(paramPath)]);
+      shouldBreak = willBreak(printed);
```
`shouldBreak = shouldBreak || willBreak(printed)`

sosukesuzuki commented:
somebody in @prettier/core can review? if approved, I'll fix conflicts.

evilebottnawi commented:
@sosukesuzuki please rebase

sosukesuzuki commented:
@evilebottnawi I fixed conflicts.

sosukesuzuki commented:
If 1.19 is released as is, the bug( #6374 ) will be shipped. So I would like somebody review and merge this PR.

evilebottnawi commented:
/cc @prettier/core need more review

sosukesuzuki commented:
I fixed conflicts. Please review someone.

evilebottnawi commented:
/cc @sosukesuzuki can you rebase
/cc @prettier/core need more review, thanks

sosukesuzuki commented:
@evilebottnawi 
I rebased, ( and hid comments resolved )

evilebottnawi commented:
/cc @prettier/core friendly ping again, it is part of `1.19`, need review

/cc @sosukesuzuki please do rebase again :disappointed:

thorn0 commented on src/language-js/printer-estree.js:
```diff
@@ -3920,26 +3920,29 @@ function printArgumentsList(path, options, print) {
-          )
-      )
-    );
+  function shouldBreakForArrowFunctionInArgments(arg, argPath) {
```
Argments -> Arguments

fisker replied:
```suggestion
  function shouldBreakForArrowFunctionInArguments(arg, argPath) {
```

fisker commented on src/language-js/printer-estree.js:
```diff
@@ -3959,7 +3962,10 @@ function printArgumentsList(path, options, print) {
     }
 
-    anyArgEmptyLine = hasEmptyLineInObjectArgInArrowFunction(arg);
+    shouldBreakForArrowFunction = shouldBreakForArrowFunctionInArgments(
```
```suggestion
    shouldBreakForArrowFunction = shouldBreakForArrowFunctionInArguments(
```

sosukesuzuki commented:
I fixed conflicts and typo. Please review someone.

evilebottnawi commented:
/cc @prettier/core need more review, thanks!

evilebottnawi commented:
/cc @prettier/core friendly ping again, please review :smile:

evilebottnawi commented:
Feel free to feedback and open a new issue if you found problems
